drivers/clk/clk-gpio.c

```diff
@@ -44,9 +45,37 @@
 struct clk_gpio {
 	struct clk_hw	hw;
 	struct gpio_desc *gpiod;
+	struct device *dev;
 };
 
 #define to_clk_gpio(_hw) container_of(_hw, struct clk_gpio, hw)
+static int clk_gpio_gate_acquire(struct clk_hw *hw)
+{
+	struct clk_gpio *clk = to_clk_gpio(hw);
+	struct device *dev = clk->dev;
+
+	clk->gpiod = devm_gpiod_get(dev, "enable", GPIOD_OUT_LOW);
+	if (IS_ERR(clk->gpiod))
+		return PTR_ERR(clk->gpiod);
```

Comment:
You still need to NULL out clk->gpiod on error (or assign the gpiod to a temporary and only set clk->gpiod on success.